machine/v1beta1/types_gcpprovider.go

```diff
@@ -146,9 +155,16 @@ type GCPMachineProviderSpec struct {
 	// confidentialCompute Defines whether the instance should have confidential compute enabled.
 	// If enabled OnHostMaintenance is required to be set to "Terminate".
 	// If omitted, the platform chooses a default, which is subject to change over time, currently that default is false.
+	// If confidentialInstanceType is configured, even if confidentialCompute is Disabled, a confidential compute instance will be configured.
```

Comment:
This feels like it could be ripe for confusion. Would it make more sense to make it so you can only set `confidentialInstanceType` when `confidentialCompute` is set to `Enabled`?

Reply:
Yes, I agree with you: it would make more sense. I've been also told the same in the ongoing discussion on https://github.com/kubernetes-sigs/cluster-api-provider-gcp/pull/1410/files#r1931842698

I will change this soon.